src/system_database.ts

```diff
@@ -684,10 +684,26 @@ export class PostgresSystemDatabase implements SystemDatabase {
     }
   }
 
-  async recordWorkflowError(workflowID: string, status: WorkflowStatusInternal): Promise<void> {
+  async recordWorkflowErrorAndDequeueIfNeeded(workflowID: string, status: WorkflowStatusInternal): Promise<void> {
     const client = await this.pool.connect();
     try {
+      await client.query('BEGIN ISOLATION LEVEL READ COMMITTED');
+      if (status.queueName) {
+        await client.query(
+          `UPDATE ${DBOSExecutor.systemDBSchemaName}.workflow_queue
+           SET completed_at_epoch_ms = $2
+           WHERE workflow_uuid = $1;`,
+          [workflowID, Date.now()],
+        );
+      } else {
+        await deleteQueuedWorkflows(client, workflowID);
```

Comment:
I assume if you want to delete queued workflows `status.queueName` is also set? Can you explain what's the logic here?

Reply:
The original implementation checks `queue.rateLimit` not queueName.

Reply:
fixed thanks